team.go

```diff
@@ -149,6 +160,49 @@ func (s *teams) Read(ctx context.Context, teamID string) (*Team, error) {
 	return t, nil
 }
 
+// TeamUpdateOptions represents the options for updating a team.
+type TeamUpdateOptions struct {
+	// For internal use only!
+	ID string `jsonapi:"primary,teams"`
+
+	// New name for the team
+	Name *string `jsonapi:"attr,name,omitempty"`
```

Comment:
If `Name` is indeed required (see comment below), then we should remove `omitempty`